### PR TITLE
P0784R7 More constexpr containers

### LINKED ISSUE #3004: [2019-07 CWG Motion 25] P0784R7 More constexpr containers

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #3004.

### CODE REVIEW

File: source/utilities.tex [hunk 1]
```diff
@@ -6764,33 +6764,43 @@
...
   template<class T>
-    void destroy_at(T* location);
+    constexpr void destroy_at(T* location);
   template<class ForwardIterator>
-    void destroy(ForwardIterator first, ForwardIterator last);
+    constexpr void destroy(ForwardIterator first, ForwardIterator last);
   template<class ExecutionPolicy, class ForwardIterator>
-    void destroy(ExecutionPolicy&& exec, // see \ref{algorithms.parallel.overloads}
+    constexpr void destroy(ExecutionPolicy&& exec,              // see \ref{algorithms.parallel.overloads}
                  ForwardIterator first, ForwardIterator last);
```
Comment (burblebee) [inline source/utilities.tex]: The parameters need to be indented to match up with the (, same for other modified functions that wrap lines - fix them all.

File: source/utilities.tex [hunk 2]
```diff
@@ -8653,15 +8664,39 @@
...
+
+\begin{itemdescr}
+\pnum
+\constraints
+The expression \tcode{::new (declval<void*>()) T(declval<Args>()...)}
+is well-formed when treated as an unevaluated operand.
+
+\effects
+Equivalent to:
+\tcode{return ::new (voidify(*location)) T(std::forward<Args>(args)...);}
```
Comment (burblebee) [inline source/utilities.tex]: This came out ugly (indented and wrapped over 2 lines).  Use codeblock instead.

File: source/expressions.tex [hunk 3]
```diff
@@ -7043,6 +7039,21 @@
 \end{codeblock}
 \end{example}
 
+\pnum
+An object \tcode{a} is said to have \defnadj{constant}{destruction} if:
```
[the remark this answers is not in the document]
Comment (zygoloid) [reply source/expressions.tex]: The paper unhelpfully wanted both paragraphs added after p4.

File: source/classes.tex [hunk 4]
```diff
@@ -2203,6 +2207,13 @@
 destructors for its base classes and its non-static data members shall have been
 implicitly defined.
 
+\pnum
+The defaulted destructor is a constexpr destructor
+if it satisfies the requirements for a constexpr destructor\iref{dcl.constexpr}.
+\begin{note}
+In particular, a trivial destructor is a constexpr destructor.
+\end{note}
```
[the remark this answers is not in the document]
Comment (zygoloid) [reply source/classes.tex]: I agree this is the wrong place, but I think moving it immediately after the wording about triviality works best. Done.

**Pull request merged**